cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -203,8 +205,9 @@ def listen_for_xblock_duplicated(sender, signal, **kwargs):
     Publish XBLOCK_DUPLICATED signals onto the event bus.
     """
     if settings.FEATURES.get("ENABLE_SEND_XBLOCK_EVENTS_OVER_BUS"):
+        topic = getattr(settings, "EVENT_BUS_XBLOCK_LIFECYLCE_TOPIC", "course-authoring-xblock-lifecycle")
         get_producer().send(
-            signal=XBLOCK_DUPLICATED, topic='xblock-duplicated',
+            signal=XBLOCK_DUPLICATED, topic=topic,
```

Comment:
@navinkarkera We seem to have 3 different events all using the same topic - which might be the legitimate usecase for someone using this feature. But wouldn't it be limiting when they want to publish them to different topics later.

What do think about using the following technique?

```
# env/base.py
EVENT_BUS_DEFAULT_XBLOCK_LIFECYCLE_TOPIC = "course-authoring-xblock"
EVENT_BUS_XBLOCK_PUBLISHED_TOPIC = "xblock-published"
EVENT_BUS_XBLOCK_DUPLICATED_TOPIC = "xblock-duplicated"
...etc.,
```

```
# env/production.py or lms.yml
EVENT_BUS_DEFAULT_XBLOCK_LIFECYCLE_TOPIC = "my-custom-topic"
EVENT_BUS_XBLOCK_PUBLISHED_TOPIC = ""
EVENT_BUS_XBLOCK_DUPLICATED_TOPIC = ""

## OR 

# env/production.py or lms.yml
EVENT_BUS_DEFAULT_XBLOCK_LIFECYCLE_TOPIC = "my-custom-topic"
EVENT_BUS_XBLOCK_PUBLISHED_TOPIC = "my-custom-topic"
EVENT_BUS_XBLOCK_DUPLICATED_TOPIC = "my-custom-topic"
```

And here we load

```
topic = getattr(settings, "EVENT_BUS_XBLOCK_DUPLICATED_TOPIC", getattr(settings, "EVENT_BUS_DEFAULT_XBLOCK_LIFECYCLE_TOPIC"))
```

This way we preserve the existing event names, thus not change the functionality and cause any regression. And at the same time, we add the option to override them?

Reply:
I purposely avoided having separate settings due to below reasons:

* These events are really new that is not used by anyone yet (that we know of)
* The intended users from whom we implemented these want to push them to the same topic, @sameenfatima78 correct me if I am wrong,
* Like mentioned, this is temporary option which is going to be removed within 2 weeks at the most, so implemented the simplest option.

@tecoholic @sameenfatima78 Still we can have separate settings which can be set to same topic name if required. What do you think?

Reply:
I agree with @navinkarkera's here. The owners of event-bus architecture suggested us to push xblock related events to the same topic and since this change is just a temporary workaround, I'd suggest we keep things simple so it can be easier to modify in the future.